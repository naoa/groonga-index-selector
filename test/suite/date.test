plugin_register expression_rewriters/index_selector

table_create expression_rewriters TABLE_HASH_KEY ShortText
column_create expression_rewriters plugin_name COLUMN_SCALAR Text

load --table expression_rewriters
[
{"_key": "index_selector", "plugin_name": "expression_rewriters/index_selector"}
]

config_set index-selector.table white_words

table_create Entries TABLE_NO_KEY
column_create Entries title COLUMN_SCALAR ShortText
column_create Entries date COLUMN_SCALAR Time

table_create white_words TABLE_PAT_KEY ShortText --default_tokenizer TokenBigram --normalizer NormalizerAuto
column_create white_words title COLUMN_INDEX|WITH_POSITION Entries title

load --table white_words
[
{"_key": "groonga"}
]

table_create Terms TABLE_PAT_KEY ShortText --default_tokenizer TokenBigram
column_create Terms title COLUMN_INDEX|WITH_POSITION Entries title

load --table Entries
[
{"title": "Groonga and MySQL", "date": "2016-01-01 00:00:00"}
]

select Entries --filter 'date == "2016-01-01 00:00:00"'
