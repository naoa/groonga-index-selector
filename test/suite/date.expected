plugin_register expression_rewriters/index_selector
[[0,0.0,0.0],true]
table_create expression_rewriters TABLE_HASH_KEY ShortText
[[0,0.0,0.0],true]
column_create expression_rewriters plugin_name COLUMN_SCALAR Text
[[0,0.0,0.0],true]
load --table expression_rewriters
[
{"_key": "index_selector", "plugin_name": "expression_rewriters/index_selector"}
]
[[0,0.0,0.0],1]
config_set index-selector.table white_words
[[0,0.0,0.0],true]
table_create Entries TABLE_NO_KEY
[[0,0.0,0.0],true]
column_create Entries title COLUMN_SCALAR ShortText
[[0,0.0,0.0],true]
column_create Entries date COLUMN_SCALAR Time
[[0,0.0,0.0],true]
table_create white_words TABLE_PAT_KEY ShortText --default_tokenizer TokenBigram --normalizer NormalizerAuto
[[0,0.0,0.0],true]
column_create white_words title COLUMN_INDEX|WITH_POSITION Entries title
[[0,0.0,0.0],true]
load --table white_words
[
{"_key": "groonga"}
]
[[0,0.0,0.0],1]
table_create Terms TABLE_PAT_KEY ShortText --default_tokenizer TokenBigram
[[0,0.0,0.0],true]
column_create Terms title COLUMN_INDEX|WITH_POSITION Entries title
[[0,0.0,0.0],true]
load --table Entries
[
{"title": "Groonga and MySQL", "date": "2016-01-01 00:00:00"}
]
[[0,0.0,0.0],1]
select Entries --filter 'date == "2016-01-01 00:00:00"'
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        1
      ],
      [
        [
          "_id",
          "UInt32"
        ],
        [
          "date",
          "Time"
        ],
        [
          "title",
          "ShortText"
        ]
      ],
      [
        1,
        1451574000.0,
        "Groonga and MySQL"
      ]
    ]
  ]
]
